src/averageAges.js

```diff
@@ -54,8 +76,46 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => {
+    const allChildren = people.find((child) => person.mother === child.name);
+
+    return allChildren;
+  })
+    .map(person => {
+      const boysMother = people.find(folk => person.mother === folk.name);
+
+      return {
+        ...person,
+        motherBorn: boysMother.born,
+      };
+    });
+
+  const childrenBoy = people.filter(person => {
+    const boy = people.find((child) => person.mother === child.name
+    && person.sex === 'm');
+
+    return boy;
+  })
+    .map(person => {
+      const boysMother = people.find(folk => person.mother === folk.name);
+
+      return {
+        ...person,
+        motherBorn: boysMother.born,
+      };
+    });
+
+  const filteredPeople = !onlyWithSon
```

Comment:
optimize your code! you do double work without considering the conditions. Put your ?: above the filters and do only the required actions!)